src/flags.rs

```diff
@@ -147,13 +153,18 @@ impl FlagContext {
         let coarse_chan_flags = &self.coarse_chan_flags[coarse_chan_range.clone()];
         let baseline_flags = self.get_baseline_flags(ant_pairs);
 
+        let fine_channel_count = self.fine_chan_flags.len();
         let chan_flags: Vec<_> = coarse_chan_flags
             .iter()
             .flat_map(|coarse_chan_flag| {
                 if *coarse_chan_flag {
-                    vec![true; self.fine_chan_flags.len()]
+                    vec![true; fine_channel_count]
                 } else {
-                    self.fine_chan_flags.clone()
+                    let mut flags = self.fine_chan_flags.clone();
+                    if self.flag_dc {
+                      flags[fine_channel_count / 2] = true;
```

Comment:
if you do this outside the `coarse_chan_flags` loop, it'll be slightly more efficient